packages/fastify-renderer/src/node/index.ts

```diff
@@ -216,7 +213,11 @@ export const build = async (fastify: FastifyInstance) => {
   })
 
   // Write a special manifest for the server side entrypoints
-  // Somewhat strangely we also use virtual entrypoints for the server side code used during SSR -- that means that in production, the server needs to require code from a special spot to get the SSR-safe version of each entrypoint. We write out our own manifest here because there's a bug in rollup or vite that errors when trying to generate a manifest in SSR mode.
+  // Somewhat strangely we also use virtual entrypoints for the server side code used during SSR --
+  // that means that in production, the server needs to require code from a special spot to get the
+  // SSR-safe version of each entrypoint. We write out our own manifest here because there's a bug
+  // in rollup or vite that errors when trying to generate a manifest in SSR mode.
+  // TODO(@ayoubelk): Figure out why we need this bit and if we can fix the issue? in rollup/vite
```

Comment:
I looked into this for a long time and could never figure it out. If you don't do this, you get really cryptic error messages, especially around there being multiple react versions. vite itself moves really fast so maybe they have fixed it up stream by now as well, but I think the move would be to get a good test suite (end to end included) going and then we can try ripping this out and seeing what breaks!